gluten-data/src/main/scala/io/glutenproject/execution/VeloxColumnarRules.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.glutenproject.execution
+
+import io.glutenproject.backendsapi.BackendsApiManager
+import io.glutenproject.columnarbatch.ArrowColumnarBatches
+import io.glutenproject.memory.arrowalloc.ArrowBufferAllocators
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution._
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+case class LoadArrowData(child: SparkPlan) extends UnaryExecNode {
+  override protected def doExecute(): RDD[InternalRow] = {
+    throw new UnsupportedOperationException(
+      "LoadArrowData does not support the execute() code path.")
+  }
+
+  override def nodeName: String = "LoadArrowData"
+
+  override def supportsColumnar: Boolean = true
+
+  override protected def doExecuteColumnar(): RDD[ColumnarBatch] = {
+    child.executeColumnar().mapPartitions {
+      itr =>
+        BackendsApiManager.getIteratorApiInstance.genCloseableColumnBatchIterator(itr.map {
+          cb => ArrowColumnarBatches.ensureLoaded(ArrowBufferAllocators.contextInstance(), cb)
+        })
+    }
+  }
+
+  override def output: Seq[Attribute] = {
+    child.output
+  }
+
+  override protected def withNewChildInternal(newChild: SparkPlan): SparkPlan =
+    copy(child = newChild)
+}
+
+object VeloxColumnarRules {
```

Comment:
Verified on internal Jenkins. Just a tiny question, can this file be put in `backends-velox` instead of `gluten-data` if it only for Velox backend?

Reply:
Hi morui, I confirmed with @zzcclp , we think:

1. currently all classes in gluten-data seems to be "for velox" . Classes "for velox+ch" are all located in gluten-core.
2. Due to 1, some classes named "GlutenXXX" in gluten-data is misleading
3. This is an unreasonable state, but we don't have the bandwidth to correct it right now

Reply:
> some classes named "GlutenXXX" in gluten-data is misleading

The classes `GlutenXXX` were added for backends that rely on Gluten to manage data conversions between Java and C++. So we put this layer (gluten-data) as an individual module out from backends-velox. But indeed we don't have another backend to use this part of code yet.

Reply:
Got it. In that case, gluten-data is designed to be a common module, right? Anyway, per previous discussion with Binwei, I will do a clean up for classes named as `GlutenXXX` and `VeloxXXX` because the prefixes are not necessary.

Reply:
> gluten-data is designed to be a common module, right?

Yes that was the original design.

Reply:
thanks @zhztheplayer for the explanation!

@rui-mo , If a class is already backend-agnostic, then I'll agree with you that the Gluten prefix is unnecessary.

However, if a class is velox-only , it's perferrable to keep the Velox prefix, so that we can easily distinguish velox-specific classes from the common classes. 

Bedi

Reply:
> I will do a clean up for classes named as GlutenXXX and VeloxXXX because the prefixes are not necessary.

@binmahone Got your point. The motivation for above action is we assume Velox related code is put in `backends-velox` module, so the `Velox` prefix becomes unnecessary. If some velox-only code exists in a common module, that's a different case which shall be rechecked. I will take a look further, thanks for pointing this out!